segmentation/utils/spatial_analysis_test.py

```diff
@@ -52,84 +78,163 @@ def make_expression_matrix(enrichment_type):
         # We create 8 cells positive for column index 2, and 8 cells positive for column index 3.
         # These are within the dist_lim in dist_mat_pos (positive enrichment distance matrix).
-        all_patient_data_pos.iloc[0:8, 2] = 1
-        all_patient_data_pos.iloc[10:18, 3] = 1
+        all_data_pos.iloc[0:8, 2] = 1
+        all_data_pos.iloc[10:18, 3] = 1
+        # We assign the two populations of cells different cell phenotypes
+        all_data_pos.iloc[0:8, 31] = 1
+        all_data_pos.iloc[0:8, 32] = "Pheno1"
+        all_data_pos.iloc[10:18, 31] = 2
+        all_data_pos.iloc[10:18, 32] = "Pheno2"
         # We create 4 cells in column index 2 and column index 3 that are also positive
         # for their respective markers.
-        all_patient_data_pos.iloc[28:32, 2] = 1
-        all_patient_data_pos.iloc[32:36, 3] = 1
+        all_data_pos.iloc[28:32, 2] = 1
+        all_data_pos.iloc[32:36, 3] = 1
+        # We assign the two populations of cells different cell phenotypes
+        all_data_pos.iloc[28:32, 31] = 1
+        all_data_pos.iloc[28:32, 32] = "Pheno1"
+        all_data_pos.iloc[32:36, 31] = 2
+        all_data_pos.iloc[32:36, 32] = "Pheno2"
         # Assign column names to columns not for markers (columns to be excluded)
-        all_patient_data_pos = all_patient_data_pos.rename(excluded_colnames, axis=1)
+        all_patient_data_pos = all_data_pos.rename(excluded_colnames, axis=1)
         return all_patient_data_pos
     elif enrichment_type == "negative":
-        all_patient_data_neg = pd.DataFrame(np.zeros((60, 31)))
+        all_data_neg = pd.DataFrame(np.zeros((60, 33)))
         # Assigning values to the patient label and cell label columns
-        all_patient_data_neg[30] = "Point8"
-        all_patient_data_neg[24] = np.arange(len(all_patient_data_neg[1])) + 1
+        all_data_neg[30] = "Point8"
+        all_data_neg[24] = np.arange(len(all_data_neg[1])) + 1
         # We create two groups of 20 cells positive for marker 1 (in column index 2)
         # and marker 2 (in column index 3) respectively.
         # The two populations are not within the dist_lim in dist_mat_neg (negative enrichment distance matrix)
-        all_patient_data_neg.iloc[0:20, 2] = 1
-        all_patient_data_neg.iloc[20:40, 3] = 1
+        all_data_neg.iloc[0:20, 2] = 1
+        all_data_neg.iloc[20:40, 3] = 1
+        # We assign the two populations of cells different cell phenotypes
+        all_data_neg.iloc[0:20, 31] = 1
+        all_data_neg.iloc[0:20, 32] = "Pheno1"
+        all_data_neg.iloc[20:40, 31] = 2
+        all_data_neg.iloc[20:40, 32] = "Pheno2"
         # Assign column names to columns not for markers (columns to be excluded)
-        all_patient_data_neg = all_patient_data_neg.rename(excluded_colnames, axis=1)
+        all_patient_data_neg = all_data_neg.rename(excluded_colnames, axis=1)
         return all_patient_data_neg
 
 
 def test_calculate_channel_spatial_enrichment():
```

Comment:
Let's add a test with multiple FOVs. See if you can do something simple, like maybe take half of one these FOVs you already generated. Don't use an exact copy, because we want to be able to check that the output is correct for each individual point, which we can't do if they're exactly identical